src/shared/components.tsx

```diff
@@ -30,25 +42,49 @@ export const Form = (props: FormHTMLAttributes<HTMLFormElement>) => {
   return <form {...props} className={classes(styles, props.className)} />;
 };
 
-export const Select = (props: SelectHTMLAttributes<HTMLSelectElement>) => {
-  const styles = `
+// eslint-disable-next-line react/display-name
+export const Select = forwardRef(
+  (
+    props: SelectHTMLAttributes<HTMLSelectElement>,
+    ref: LegacyRef<HTMLSelectElement>
+  ) => {
+    const styles = `
     block w-full rounded-md border-0 py-1.5 px-1.5 text-black shadow-sm ring-1 ring-inset ring-gray-300
     sm:max-w-xs sm:text-sm sm:leading-6
     ${focusStyles}
   `;
-  return <select {...props} className={classes(styles, props.className)} />;
-};
+    return (
+      <select
+        {...props}
+        className={classes(styles, props.className)}
+        ref={ref}
+      />
+    );
+  }
+);
 
 export const Label = (props: LabelHTMLAttributes<HTMLLabelElement>) => {
   const styles = "block text-sm font-medium leading-6 text-white text-justify";
   return <label {...props} className={classes(styles, props.className)} />;
 };
 
-export const Input = (props: InputHTMLAttributes<HTMLInputElement>) => {
-  const styles = `
+// eslint-disable-next-line react/display-name
+export const Input = forwardRef(
+  (
+    props: InputHTMLAttributes<HTMLInputElement>,
+    ref: LegacyRef<HTMLInputElement>
+  ) => {
```

Comment:
can we make a util function to abstract this logic and avoid duplicating this to every component we have?

We should also take care of this eslint issue